includes/shortcodes/class-wc-shortcode-my-account.php

```diff
@@ -39,7 +39,7 @@ public static function output( $atts ) {
 			return;
 		}
 
-		if ( ! is_user_logged_in() || isset( $wp->query_vars['lost-password'] ) ) {
+		if ( ! is_user_logged_in() && isset( $wp->query_vars['lost-password'] ) ) {
```

Comment:
It introduces a bug, now there's no more a login form, everyone that tries to access `/my-account/` will see the sidebar and all items like they are logged in.